ISISScientificComputing/autoreduce
# Skip when beam off

### Summary of work
* Added validation module that performs various validation checks on the data
* Added `testing_data` module that houses some `.nxs` files used in testing
* Added to the data archive creator functionality to ensure that it can handle copying "real" files (not just mocked empty files) into a fake data archive. 
* Ensured the system tests still pass by using a valid data file (note these cannot be mocked as the queue processor code is started as an  external service from within the system test) 

### How to test your work
* Code review
* Checkout this branch on the development compute node and restart the queue processors
* Resubmit some runs with 0 beam current
  * Ensure these are skipped
* Resubmit some runs with 0 or less as an RB number ( or a string RB)
  * Ensure these are skipped
* Resubmit normal runs we expect to be reduced
  * Ensure these actually get reduced 


Fixes #452

*Release notes have been updated*
